src/Engine/ProtoCore/DSASM/Executive.cs

```diff
@@ -3921,7 +3921,7 @@ push that
                     if (Constants.kGlobalScope == classId)
                     {
                         procName = exe.procedureTable[blockId].procList[procId].name;
-                        CodeBlock codeblock = core.GetCodeBlock(core.CodeBlockList, blockId);
+                        CodeBlock codeblock = ProtoCore.Utils.CoreUtils.GetCodeBlock(exe.CodeBlocks, blockId);
```

Comment:
Do you mean it is unsafe to use `CodeBlockList[blockId]` and `CompleteCodeBlockList[blockId]`? If yes, we need to remove all of them.